File: 1.TwoNumAddUptoK/Solution.py
from typing import List

# Solution Class
class Solution:
    def solve(nums:List[int], k:int) -> bool:
        check = []
        for i in nums:
            diff = k-i
            if diff in check:
                return True
            check.append(i)
        return False

# Main  
nums = list(map(int,input().split()))
k = int(input())
result = Solution.solve(nums,k)
print(result) 

File: 1.TwoNumAddUptoK/Solution.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
    bool solve(std::vector<int> &nums, int k)
    {
        std::vector<int> check;
        for (int i : nums)
        {
            int diff = k - i;
            if (std::find(check.begin(), check.end(), diff) != check.end())
            {
                return true;
            }
            check.push_back(i);
        }
        return false;
    }
};

int main()
{
    string input;
    getline(cin, input);

    vector<int> nums;
    stringstream ss(input);
    string token;
    while (getline(ss, token, ' '))
    {
        int num = stoi(token);
        nums.push_back(num);
    }
    int k;
    std::cin >> k;
    bool result = Solution().solve(nums, k);
    std::cout << result;
    return 0;
}
